### PR TITLE
feat(blocks,store): use schema in block model

### PR DESCRIPTION
**Pull request by himself65**
Related: https://github.com/toeverything/blocksuite/issues/151

### CODE REVIEW

File: packages/store/src/base.ts [hunk 1]
```diff
@@ -2,6 +2,29 @@ import type { Page } from './workspace/index.js';
...
+  version: z.number(),
+  schema: z.object({
+    flavour: z.string(),
+    tag: z.string(),
+    type: z.string().optional(),
+    defaultProps: z.record(z.any()),
+  }),
+});
+
+const CodeBlockSchema = {
```
Comment (doodlewind) [inline packages/store/src/base.ts:17]: I think we should not put the per-block definition in the `store` package.

Comment (himself65) [reply packages/store/src/base.ts]: just a example here

**Comment by himself65** [comment]
blocking because there is need to discussion about the design

**Comment by himself65** [comment]
After discussion. This code needs a little change

File: packages/store/src/base.ts [hunk 2]
```diff
@@ -10,22 +10,15 @@ const TagSchema = z.object({
...
 
-const StateSchema = z.any();
-
 export const BlockSchema = z.object({
   version: z.number(),
   model: z.object({
     flavour: FlavourSchema,
     tag: TagSchema,
-    state: StateSchema,
+    state: z.function().returns(z.record(z.any())),
```
Comment (himself65) [inline packages/store/src/base.ts:18]: we use `() => state` here to avoid the Inconsistent state

**Review by doodlewind — APPROVED** [review]
LGTM, a good start!

**Pull request merged**